packages/apps/spaces/app/src/components/Providers/SocketProvider.tsx

```diff
@@ -0,0 +1,33 @@
+'use client';
+
+import { useState, useEffect, createContext } from 'react';
+
+import { Socket } from 'phoenix';
+
+import { useEnv } from '@shared/hooks/useEnv';
+
+const PhoenixSocketContext = createContext<{ socket: Socket | null }>({
+  socket: null,
+});
+
+const PhoenixSocketProvider = ({ children }: { children: React.ReactNode }) => {
+  const env = useEnv();
+  const [socket, setSocket] = useState<Socket | null>(null);
+  const socketPath = `${env.REALTIME_WS_PATH}/socket`;
+
+  useEffect(() => {
+    const socket = new Socket(socketPath, { params: { token: '123' } });
+    socket.connect();
+    setSocket(socket);
+  }, [socketPath]);
+
+  if (!socket) return null;
```

Comment:
let's return children here, in case of any socket malfunction user could still use rest of the app